pkg/cluster/k8sres.go

```diff
@@ -872,9 +880,14 @@ func (c *Cluster) generateStatefulSet(spec *acidv1.PostgresSpec) (*appsv1.Statef
 	}
 
 	// generate sidecar containers
-	if sidecarContainers, err = generateSidecarContainers(sideCars, volumeMounts, defaultResources,
-		c.OpConfig.SuperUsername, c.credentialSecretName(c.OpConfig.SuperUsername), c.logger); err != nil {
-		return nil, fmt.Errorf("could not generate sidecar containers: %v", err)
+	if sideCars != nil && len(sideCars) > 0 {
+		if c.OpConfig.EnableSidecars != nil && !(*c.OpConfig.EnableSidecars) {
+			c.logger.Warningf("sidecars in use but globally disabled - next stateful set creation would fail")
```

Comment:
"in use" as "specified in the manifest" ?

Reply:
this code won't be reached on cluster creation because `createStatefulSet` would throw an error with this wording ("specified but disabled"). Therefore, during snyc and update I can assume that sidecars and initContainers are in use or running.

Reply:
sure it is just that phrases "in use" and "globally" make sense for people familiar with operator internals and this feature in particular but not necessarily to those reading the logs

Reply:
updated the messages. Should be clearer now.